src/main/java/uk/nhs/tis/sync/job/PostFundingStatusSyncJob.java

```diff
@@ -100,29 +100,32 @@ protected int convertData(Set<Post> entitiesToSave, List<EntityData> entityData,
       Post post = entityManager.find(Post.class, entity.getEntityId());
 
       if (post != null) {
-        Set<PostFunding> fundings = post.getFundings();
-
-        if (fundings.size() > 1) {
-          boolean allEndDatesYesterday = true;
-          for (PostFunding funding : fundings) {
-            if(!yesterday.equals(funding.getEndDate())) {
-              allEndDatesYesterday = false;
-              break;
-            }
-          }
-          if(allEndDatesYesterday) {
-            post.setFundingStatus(Status.INACTIVE);
-          }
-          entitiesToSave.add(post);
-        } else if (fundings.size() == 1) {
-          post.setFundingStatus(Status.INACTIVE);
-          entitiesToSave.add(post);
-        }
+        processPostFundingStatus(entitiesToSave, yesterday, post);
       }
     }
     return entities - entitiesToSave.size();
   }
 
+  private void processPostFundingStatus(Set<Post> entitiesToSave, LocalDate yesterday, Post post) {
+    Set<PostFunding> fundings = post.getFundings();
+    if (!fundings.isEmpty()) {
+      boolean allEndDatesYesterday = checkAllEndDatesYesterday(fundings, yesterday);
+      if (allEndDatesYesterday) {
+        post.setFundingStatus(Status.INACTIVE);
+      }
+      entitiesToSave.add(post);
```

Comment:
I don't understand why do you still need to save the entity if its status is not updated.

Reply:
This line is now moved inside the if block